pkg/scalers/azure/azure_app_insights.go

```diff
@@ -0,0 +1,150 @@
+package azure
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/azure/auth"
+
+	kedav1alpha1 "github.com/kedacore/keda/v2/apis/keda/v1alpha1"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const (
+	appInsightsResource = "https://api.applicationinsights.io"
+)
+
+type AppInsightsInfo struct {
+	ApplicationInsightsID string
+	TenantID              string
+	MetricID              string
+	AggregationTimespan   string
+	AggregationType       string
+	Filter                string
+	ClientID              string
+	ClientPassword        string
+}
+
+type ApplicationInsightsMetric struct {
+	Value map[string]interface{}
+}
+
+var azureAppInsightsLog = logf.Log.WithName("azure_app_insights_scaler")
+
+func toISO8601(time string) (string, error) {
+	timeSegments := strings.Split(time, ":")
+	if len(timeSegments) != 2 {
+		return "", fmt.Errorf("invalid interval %s", time)
+	}
+
+	hours, herr := strconv.Atoi(timeSegments[0])
+	minutes, merr := strconv.Atoi(timeSegments[1])
+
+	if herr != nil || merr != nil {
+		return "", fmt.Errorf("errors parsing time: %v, %v", herr, merr)
+	}
+
+	return fmt.Sprintf("PT%02dH%02dM", hours, minutes), nil
+}
+
+func getAuthConfig(info AppInsightsInfo, podIdentity kedav1alpha1.PodIdentityProvider) auth.AuthorizerConfig {
+	if podIdentity == "" || podIdentity == kedav1alpha1.PodIdentityProviderNone {
+		config := auth.NewClientCredentialsConfig(info.ClientID, info.ClientPassword, info.TenantID)
+		config.Resource = appInsightsResource
+		return config
+	}
+
+	config := auth.NewMSIConfig()
+	config.Resource = appInsightsResource
+	return config
+}
+
+func extractAppInsightValue(info AppInsightsInfo, metric ApplicationInsightsMetric) (int32, error) {
+	if _, ok := metric.Value[info.MetricID]; !ok {
+		return -1, fmt.Errorf("metric named %s not found in app insights response", info.MetricID)
+	}
+
+	var err error
+	floatVal := 0.0
+	if val, ok := metric.Value[info.MetricID].(map[string]interface{})[info.AggregationType]; ok {
+		if val == nil {
+			err = fmt.Errorf("metric %s was nil for aggregation type %s", info.MetricID, info.AggregationType)
+		} else {
+			floatVal = val.(float64)
+		}
+	} else {
+		err = fmt.Errorf("metric %s did not containe aggregation type %s", info.MetricID, info.AggregationType)
+	}
+
+	if err != nil {
+		return -1, err
+	}
```

Comment:
Should we avoid these else?
```suggestion
	if val, ok := metric.Value[info.MetricID].(map[string]interface{})[info.AggregationType]; ok {
		if val == nil {
			return -1, fmt.Errorf("metric %s was nil for aggregation type %s", info.MetricID, info.AggregationType)
		} 
		floatVal = val.(float64)
	} else {
		return -1, fmt.Errorf("metric %s did not containe aggregation type %s", info.MetricID, info.AggregationType)
	}
```

Reply:
Yes. Thanks.